src/placement/tests/place_obj_common.c

```diff
@@ -817,10 +818,20 @@ extend_test_pool_map(struct pool_map *map,
 	rc = gen_pool_buf(map, &map_buf, map_version, ndomains, nnodes,
 			  ntargets, domains, target_uuids, rank_list, NULL,
 			dss_tgt_nr);
-	assert_success(rc);
+	if (rc != 0)
+		D_GOTO(out_map, rc);
```

Comment:
@liw seems like gen_pool_buf should have already freed the map_buf (pool_buf_free()) before returning nonzero, so I guess this would be ok to remove. Is that what you are suggesting?

Reply:
Yes. If rc != 0, we should not read map_buf's value and compare it with NULL. And we also do not need to.

Reply:
In other words, I think the general rule for a function f(some_type *out) that output some_type is that if f returns an error, out may point to an undefined value of some_type, unless f's documentation explicitly says otherwise. So if f allocates memory for out internally, it must free the memory upon errors. The caller therefore should not, and does not need to, read *out upon errors.